modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/state/DefaultResourceOwnerOAuthContext.java

```diff
@@ -90,7 +106,33 @@ public String getResourceOwnerId() {
     return resourceOwnerId == null ? DEFAULT_RESOURCE_OWNER_ID : resourceOwnerId;
   }
 
+  /**
+   * @deprecated use {@link #getRefreshUserOAuthContextLock(String, LockFactory)} instead of the lock provided here
+   */
+  @Deprecated
   public void setRefreshUserOAuthContextLock(Lock refreshUserOAuthContextLock) {
     this.refreshUserOAuthContextLock = refreshUserOAuthContextLock;
   }
+
+  @Override
+  public DancerState getDancerState() {
+    return dancerState != null ? dancerState : HAS_TOKEN;
```

Comment:
shouldn't NO_TOKEN be the default?

Reply:
By default, new instances are built with NO_TOKEN. This condition here accounts for the case where the object was generated by a previous version where this attribute wasn't there. In that case, the equivalent state for that is HAS_TOKEN.